src/core/component.ts

```diff
@@ -103,4 +113,16 @@ const DxComponent: VueConstructor = Vue.extend({
     }
 });
 
-export default DxComponent;
+const DxExtensionComponent: VueConstructor = Vue.extend({
+    extends: DxComponent,
+    methods: {
+        $_createWidget() {
+            (this as any).$isExtension = true;
+        },
+        $createInstance(element: any) {
```

Comment:
It seems that the method name should start with `$_` prefix.
What about `$_instantiate` or `$_create` ?